### PR TITLE
Improve `nagios.loadhosts`

### PR DESCRIPTION
**Pull request by alexmuller**
My typical use case is searching for unhandled problems in Icinga and running a task on all machines which match.

Instead of having to find an Icinga URL that outputs JSON it would be more useful for me to only provide a search string.

This commit changes from using the `requests` library locally to running curl on the remote machine.

I've also added the `@hosts` decorator to this task so that a machine to run it on doesn't have to be provided.

---

Feedback on whether this is a good idea is welcome.

### CODE REVIEW

**Comment by mattbostock** [comment]
One issue this is that it's not possible to preview the hosts that will be targeted (whereas it was possible when using the JSON URL). I wonder if it should prompt to confirm the list of hosts first.

**Comment by alexmuller** [comment]
I think it will still prompt to confirm with both methods.

**Comment by mattbostock** [comment]
@alexmuller I didn't spot `print` call at the bottom, though my concern is more about confirming that it's okay to proceed.

I guess if the wrong hosts are selected, there'd be time to hit Ctrl+C before the command runs, but I'm not sure we want to rely on that.

I'm cautious due to the issues we saw with search string anchoring when automating overnight reboots.

**Comment by alexmuller** [comment]
There's already a prompt in place which is still there :)

```
[HOST] 11:11:24 ~/govuk/fabric-scripts (try-to-improve-loadhosts) $ fab production vm.nagios.loadhosts:reboot do:uptime
[alert.cluster] Executing task 'vm.nagios.loadhosts'

Selected hosts:
  - api-redis-1.api
  - asset-slave-1.backend
  - graphite-1.management
  - logs-redis-1.management
  - logs-redis-2.management
  - monitoring-1.management
  - mysql-master-1.backend
  - mysql-slave-2.backend
  - postgresql-standby-1.backend
  - redis-1.backend
  - whitehall-mysql-slave-2.backend

Type 'yes' to confirm: 
```

**Comment by mattbostock** [comment]
Apologies, I missed that beneath the fold in the diff. Thanks @alexmuller.

File: nagios.py [hunk 1]
```diff
@@ -60,18 +60,26 @@ def schedule_downtime(host,minutes='20'):
...
-        https://nagios.example.com/cgi-bin/icinga/status.cgi?search_string=puppet+last+run&limit=0&start=1&servicestatustypes=29
+    Optionally takes a search string. If provided, searches for all unhandled problems.
+    If not provided, prompts for a URL like:
+        https://alert.cluster/cgi-bin/icinga/status.cgi?search_string=puppet+last+run&limit=0&start=1&servicestatustypes=29
     """
 
-    url = prompt("Icinga URL (jsonformat): ")
-    resp = requests.get(url, verify=False)
+    if search_string:
+        url = 'https://alert.cluster/cgi-bin/icinga/status.cgi?search_string={0}&allunhandledproblems&jsonoutput'.format(search_string)
```
Comment (mattbostock) [inline nagios.py]: Do we need to URL-encode the search string?

Comment (alexmuller) [inline nagios.py]: That was outside the scope of what I needed but also doesn't hurt. Done!

**Comment by mattbostock** [comment]
I should have asked this earlier, but what's the reason for switching the requests library for Curl?

**Comment by alexmuller** [comment]
It's because the requests library runs locally on your laptop so it needs the hostname of the monitoring instance to connect to. I think a better way to do it is running the query on the remote machine which means using curl (because it's one of the few things installed).

**Comment by mattbostock** [comment]
Thanks @alexmuller. It would be good to clarify that in the commit message and also explain why `--insecure` is used (as the SSL certificate is not configured for `alert.cluster`).

**Comment by alexmuller** [comment]
I've put that in the commit message for 4272b9d962bbcc525996baf7b0ee5c8531869066.

**Comment by mattbostock** [comment]
:+1:

**Pull request merged**